src/common/transformations/src/transformations/symbolic_transformations/dereshape_matmul.cpp

```diff
@@ -0,0 +1,336 @@
+    } else {
+        // no Concat and it doesn't matter if BEA is present -- just delete reshape
+        ov::replace_output_update_name(reshape_output, original_reshape->input_value(0));
+    }
+}
+}  // namespace
+
+#define IN_RESHAPE                                                                                            \
+    pattern::wrap_type<op::v1::Reshape>(pattern::op::as_value_predicate([](std::shared_ptr<Node> n) -> bool { \
+        return pattern::consumers_count(1)(n->output(0)) && reshape_keeps_last_two_dims(n);                   \
+    }));
+
+#define SCALAR_INPUT                                                                        \
+    pattern::any_input([](ov::Output<Node> out) {                                           \
+        return out.get_partial_shape().is_static() && ov::shape_size(out.get_shape()) == 1; \
+    });
+
+ov::pass::DeReshapeMatMul::DeReshapeMatMul() {
+    MATCHER_SCOPE(DeReshapeMatMul);
+    // BEGIN: symmetrical patterns for MatMul inputs
+
+    // lhs of MatMul
+    auto lhs_reshape = IN_RESHAPE;
+
+    auto lhs_concat_0 = pattern::wrap_type<op::v0::Concat>({pattern::any_input(), lhs_reshape}, concat_predicate);
+    auto lhs_concat_1 = pattern::wrap_type<op::v0::Concat>({lhs_reshape, pattern::any_input()}, concat_predicate);
+    auto lhs_concat = std::make_shared<pattern::op::Or>(OutputVector{lhs_concat_0, lhs_concat_1});
+
+    auto lhs_reshape_or_concat = std::make_shared<pattern::op::Or>(OutputVector{lhs_reshape, lhs_concat});
+
+    auto lhs_bea_scalar = SCALAR_INPUT;
+    auto lhs_bea = pattern::wrap_type<op::util::BinaryElementwiseArithmetic>({lhs_reshape_or_concat, lhs_bea_scalar},
+                                                                             pattern::consumers_count(1));
+
+    auto lhs_bea_or_concat = std::make_shared<pattern::op::Or>(OutputVector{lhs_reshape_or_concat, lhs_bea});
+
+    // rhs of MatMul
+    auto rhs_reshape = IN_RESHAPE;
+
+    auto rhs_concat_0 = pattern::wrap_type<op::v0::Concat>({pattern::any_input(), rhs_reshape}, concat_predicate);
+    auto rhs_concat_1 = pattern::wrap_type<op::v0::Concat>({rhs_reshape, pattern::any_input()}, concat_predicate);
+    auto rhs_concat = std::make_shared<pattern::op::Or>(OutputVector{rhs_concat_0, rhs_concat_1});
+
+    auto rhs_reshape_or_concat = std::make_shared<pattern::op::Or>(OutputVector{rhs_reshape, rhs_concat});
+
+    auto rhs_bea_scalar = SCALAR_INPUT;
+    auto rhs_bea = pattern::wrap_type<op::util::BinaryElementwiseArithmetic>({rhs_reshape_or_concat, rhs_bea_scalar},
+                                                                             pattern::consumers_count(1));
+
+    auto rhs_bea_or_concat = std::make_shared<pattern::op::Or>(OutputVector{rhs_reshape_or_concat, rhs_bea});
+    // END: symmetrical patterns for MatMul inputs
+
+    auto matmul =
+        pattern::wrap_type<op::v0::MatMul>({lhs_bea_or_concat, rhs_bea_or_concat}, pattern::consumers_count(1));
```

Comment:
is it possible to use just 1 pattern, e. g lhs_bea_or_concat? and delete rhs_bea_or_concat [pattern
pattern::wrap_type<op::v0::MatMul>({lhs_bea_or_concat, lhs_bea_or_concat}